src/server/pfs/server/driver.go

```diff
@@ -2344,3 +2342,18 @@ type branchCommit struct {
 	commit *pfs.Commit
 	branch *pfs.Branch
 }
+
+func appendSubvenance(commitInfo *pfs.CommitInfo, subvCommitInfo *pfs.CommitInfo) {
+	if subvCommitInfo.ParentCommit != nil {
+		for _, subvCommitRange := range commitInfo.Subvenance {
+			if subvCommitRange.Upper.ID == subvCommitInfo.ParentCommit.ID {
+				subvCommitRange.Upper = subvCommitInfo.Commit
```

Comment:
I thought of a potential problem with this approach, that I wanted to check: input commits will have several repos appearing in their subvenance if they're in a path with more than one pipeline in it:
```
A
  \
    >-- X --- Y 
  /
B
```
> commits in A will have commits in both `X` and `Y` in their subvenance.

If we're not careful, we'll end up with a worse representation than we had before:
```
A/1 {
  Subvenance {
    Range { X/1, X/1 },
    Range { Y/1, Y/1 },
    Range { X/2, X/2 },
    Range { Y/2, Y/2},
    ...
  }
}
```

I think we need to search through `commitInfo.Subvenance` for all of the ranges in the same repo as `subvCommitInfo`, so that instead we end up with:
```
A/1 {
  Subvenance {
    Range { X/1, X/5 },
    Range { Y/1, Y/5 },
    ...
  }
}
```